vignettes/intro_to_allotalot.Rmd

```diff
@@ -16,159 +16,165 @@ library(allotalot)
+minWt
+q1Wt
+medWt
+meanWt
+q3Wt
+maxWt
+```
 
-The simple Python function `coords()` returns a static list of latitude, longitude coordinates. 
+If the left-hand side and right-hand sides don't match,
+an error is raised.
+This guards against missing or unexpected values.
 
-```{python}
-def coords():
-  '''A function which returns a list of 2 numbers (coordinates)'''
-  return [38.061944, -122.643889]
-  
-lat, lng = coords()
-print(lat)
-print(lng)
+```{r, error=TRUE}
+{ stg1 : stg2 : stg3 } %<-% list("Moe", "Donald")
 ```
 
-In the example, the list of values returned by `coords()` is unpacked into the variables `lat` and `lng`. This instance of parallel assignment is commonly referred to as *unpacking assignment* or *destructuring assignment*.
-
-Here is the same coordinates example redone in R.
+```{r, error=TRUE}
+{ stg1 : stg2 : stg3 } %<-% list("Moe", "Larry", "Curley", "Donald")
+```
 
-```{R}
-#' Get a Pair of Coordinates
-#' 
-#' A function which returns a list of 2 numeric values.
-#' 
-coords <- function() {
+### Unpacking a returned value
+A common use-case is when a function returns a list of values
+and you want to extract the individual values.
+In this example, the list of values returned by `coordsList()` is unpacked
+into the variables `lat` and `lng`.
+
+```{r}
+#
+# A function which returns a list of 2 numeric values.
+# 
+coordsList <- function() {
   list(38.061944, -122.643889)
 }
 
-{lat: lng} %<-% coords()
+{ lat : lng } %<-% coordsList()
 lat
 lng
 ```
 
-Notice the use of the colon and braces on the left-hand side of the assignment expression. Based on the left-hand side structure, the assignment operator expects `coords()` to return a list of two values. Those two values are then assigned to `lat` and `lng`.
+As in the earlier example,
+colon and braces on the left-hand side of the assignment
+tell the assignment operator to expect a list.
 
-Let's try rerunning the `coords` example, but this time without the braces.
+In this example, we call a function that returns a vector.
+Now we omit the braces.
 
-```{R, include = FALSE} 
-rm(lat)
-rm(lng)
-```
+```{r}
+#
+# Convert cartesian coordinates to polar
+#
+toPolar = function(x, y) {
+  c(sqrt(x^2 + y^2), atan(y/x))
+}
 
-```{R, error = TRUE}
-lat: lng %<-% coords()
-lat
-lng
+radius : angle %<-% toPolar(12, 5)
+radius
+angle
 ```
 
-To unpack a list of values, the left-hand side must be enclosed in braces. Without the braces the assignment operator expects and tries to unpack a vector of values. If we create a new function `coords2` which returns a vector, the error is fixed.
+### Example: Intercept and slope of regression
+We can directly unpack the coefficients of a simple linear regression
+into the intercept and slope.
 
-```{R}
-#' Get Another Pair of Coordinates
-#' 
-#' A function which returns a numeric vector of length 2, the coordinates.
-#'
-coords2 <- function() {
-  c(-36.85, 174.77)
-}
+```{r}
+inter : slope %<-% coef(lm(mpg ~ cyl, data = mtcars))
```

Comment:
Thank you, thank you, thank you (!!) for coming up with better examples. I did not feel like I was doing unpacking justice.